Mlem/Models/Trackers/Inbox/InboxItem.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  InboxItem.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2023-06-26.
+//
+
+import Foundation
+import SwiftUI
+
+/// Enumeration of content models that can appear in the inbox
+enum InboxItem {
```

Comment:
Do we need this type? There seems to be a lot of logic surrounding this. Perhaps just using the same type for the parent and child type would be better? Something like this? It does require using base classes rather than protocols though

```swift
class ContentIdentifiable {
    var uid: ContentModelIdentifier
}

class InboxItem: ContentIdentifiable {
    var published: Date
    // etc
}

class MessageModel: InboxItem {
}

class ParentTracker<Item: ContentIdentifiable>: ObservableObject, ParentTrackerProtocol {
}

class InboxTracker: ParentTracker<InboxItem> {
}
```

Instead of using a switch statement to determine types view-side, we could use `as?` statements

Reply:
Updated per Slack